coderd/externalauth/externalauth.go

```diff
@@ -190,6 +190,21 @@ validate:
 		}
 		externalAuthLink = updatedAuthLink
 	}
+
+	// Update the associated users github.com username if the token is for github.com.
+	if IsGithubDotComURL(c.AuthCodeURL("")) && user != nil {
+		err = db.UpdateUserGithubComUserID(ctx, database.UpdateUserGithubComUserIDParams{
+			ID: externalAuthLink.UserID,
+			GithubComUserID: sql.NullInt64{
+				Int64: user.ID,
+				Valid: true,
+			},
+		})
+		if err != nil {
+			return externalAuthLink, xerrors.Errorf("update user github com user id: %w", err)
+		}
+	}
```

Comment:
This code executes every `RefreshToken`, but the user's github ID will not change in a refresh. Ideally we only do this check/update on auth in https://github.com/coder/coder/blob/2b8000d03aca6a3edcfa74623cdf91313528911f/coderd/externalauth.go#L234-L234. It does look like it would take some work to plumb the correct github user values to the route, so I understand why it is here.


---

If we keep this here, let's just reduce the times it is called. Since this `RefreshToken` is called quite often just to "check" the token.

Move this into the if statement above with `db.UpdateExternalAuthLink`. This would at least only update the github id if there is a new token issued. Since the old token already called `UpdateUserGithubComUserID`, it is pointless to do it multiple times.